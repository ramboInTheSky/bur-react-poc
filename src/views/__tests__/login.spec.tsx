import React from 'react';
import { shallow } from 'enzyme';

import Login, { Form } from '../login';

const mockDispatch = jest.fn();
const mockLogin = jest.fn();

jest.mock('../../state/actions', () => () => ({
    login: mockLogin,
    logout: () => {},
    foo: 'bar',
}));

jest.mock('../../state', () => ({
  useStateValue: () => [
    {
      user: undefined,
      forms: { username: 'foo', password: 'bar' },
    },
    mockDispatch,
  ],
}));

describe(Login, () => {
  const props = {
    history: {
      location: {
        search: '?redirect=baz',
      },
    },
  };

  it('should render correctly', () => {
    expect(shallow(<Login {...props} />)).toMatchSnapshot();
  });

  it('should fire correct dispatch on input change', () => {
    const component = shallow(<Login {...props} />);
    const password = component.find('[type="password"]');
    password.simulate('change', { target: { value: 'baz' } });
    expect(mockDispatch).toHaveBeenCalledWith({ payload: 'baz', type: 'updatePassword' });
  });

  it('should attempt login on form submit', () => {
    const component = shallow(<Login {...props} />);
    const form = component.find(Form);
    form.simulate('submit', { preventDefault: () => {} });
    expect(mockLogin).toHaveBeenCalledWith('foo', 'bar', 'baz');
  });
});
