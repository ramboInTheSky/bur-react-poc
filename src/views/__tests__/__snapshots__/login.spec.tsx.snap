// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render correctly 1`] = `
<styled.div>
  <Header
    onLogout={[Function]}
    title="Login"
  />
  <styled.div>
    <styled.form
      onSubmit={[Function]}
    >
      <TextInput
        onChange={[Function]}
        placeholder="Email"
        value="foo"
      />
      <TextInput
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value="bar"
      />
      <Button
        label="Login"
        type="submit"
      />
    </styled.form>
  </styled.div>
</styled.div>
`;
